plugins/entityreference/behavior/OgBehaviorHandler.class.php

```diff
@@ -122,11 +122,7 @@ public function OgMembershipCrud($entity_type, $entity, $field, $instance, $lang
     foreach ($items as $item) {
       $gid = $item['target_id'];
 
-      // Must provide correct state in the event that approval is required.
-      if (empty($item['state']) && $entity_type == 'user' && !og_user_access($group_type, $gid, 'subscribe without approval', $entity) && !og_user_access($group_type, $gid, 'administer group')) {
```

Comment:
Removing this seems wrong. I guess you tested a specific case of adding members via UI, but this is about creating membership via the field API.

I believe the existing code should be reverted